sdk/schemaregistry/azure-data-schemaregistry-jsonschema/src/main/java/com/azure/data/schemaregistry/jsonschema/SchemaRegistryJsonSchemaSerializerBuilder.java

```diff
@@ -0,0 +1,136 @@
+
+    /**
+     * Specifies schema group for interacting with Azure Schema Registry service.
+     *
+     * If auto-registering schemas, schema will be stored under this group. If not auto-registering, serializer will
+     * request schema ID for matching data schema under specified group.
+     *
+     * @param schemaGroup Azure Schema Registry schema group
+     *
+     * @return updated {@link SchemaRegistryJsonSchemaSerializerBuilder} instance
+     */
+    public SchemaRegistryJsonSchemaSerializerBuilder schemaGroup(String schemaGroup) {
+        this.schemaGroup = schemaGroup;
+        return this;
+    }
+
+    /**
+     * JSON serializer adapter to use. If none is supplied, the default is used.
+     *
+     * @param serializerAdapter Serializer to use.
+     *
+     * @return updated {@link SchemaRegistryJsonSchemaSerializerBuilder} instance.
+     */
+    public SchemaRegistryJsonSchemaSerializerBuilder serializer(SerializerAdapter serializerAdapter) {
+        this.serializerAdapter = serializerAdapter;
+        return this;
+    }
+
+    /**
+     * The {@link SchemaRegistryAsyncClient} to use to interact with the Schema Registry service.
+     *
+     * @param schemaRegistryAsyncClient The {@link SchemaRegistryAsyncClient}.
+     *
+     * @return updated {@link SchemaRegistryJsonSchemaSerializerBuilder} instance.
+     */
+    public SchemaRegistryJsonSchemaSerializerBuilder schemaRegistryClient(
+        SchemaRegistryAsyncClient schemaRegistryAsyncClient) {
+        this.schemaRegistryAsyncClient = schemaRegistryAsyncClient;
+        return this;
+    }
+
+    /**
+     * A JSON schema aware class that can generate and validate JSON schema for objects.
+     *
+     * @param jsonSchemaGenerator The JSON schema generator.
+     *
+     * @return updated {@link SchemaRegistryJsonSchemaSerializerBuilder} instance.
+     */
+    public SchemaRegistryJsonSchemaSerializerBuilder jsonSchemaGenerator(JsonSchemaGenerator jsonSchemaGenerator) {
+        this.jsonSchemaGenerator = jsonSchemaGenerator;
+        return this;
+    }
+
+    /**
+     * Creates a new instance of Schema Registry serializer.
+     *
+     * @return A new instance of {@link SchemaRegistryJsonSchemaSerializer}.
+     *
+     * @throws NullPointerException if {@link #schemaRegistryClient(SchemaRegistryAsyncClient)} or
+     *     {@link #jsonSchemaGenerator(JsonSchemaGenerator)} is {@code null}.
+     * @throws IllegalStateException if {@link #autoRegisterSchemas(boolean)} is true but
+     *     {@link #schemaGroup(String)} is not set.
+     */
+    public SchemaRegistryJsonSchemaSerializer buildSerializer() {
+        if (Objects.isNull(schemaRegistryAsyncClient)) {
+            throw LOGGER.logExceptionAsError(new NullPointerException("'schemaRegistryAsyncClient' cannot be null."));
+        }
+
+        if (Objects.isNull(jsonSchemaGenerator)) {
+            throw LOGGER.logExceptionAsError(new NullPointerException("'jsonSchemaGenerator' cannot be null."));
```

Comment:
`JsonSchemaGenerator` is an abstract class and there are no concrete implementations of this type in this library. Should users extend from this type and provide their own implementation?

Reply:
This is correct. We (the SDK) don't want to bring in any external dependencies like networknt.